cmd/os-reliable.go

```diff
@@ -22,6 +22,50 @@ import (
 	"path"
 )
 
+// Wrapper functions to os.RemoveAll, which calls reliableRemoveAll
+// this is to ensure that if there is a racy parent directory
+// create in between we can simply retry the operation.
+func removeAll(dirPath string) (err error) {
+	if dirPath == "" {
+		return errInvalidArgument
+	}
+
+	if err = checkPathLength(dirPath); err != nil {
+		return err
+	}
+
+	if err = reliableRemoveAll(dirPath); err != nil {
+		// File path cannot be verified since one of the parents is a file.
+		if isSysErrNotDir(err) {
+			return errFileAccessDenied
+		} else if isSysErrPathNotFound(err) {
+			// This is a special case should be handled only for
+			// windows, because windows API does not return "not a
+			// directory" error message. Handle this specifically here.
+			return errFileAccessDenied
+		}
+	}
+	return err
+}
+
+// Reliably retries os.RemoveAll if for some reason os.RemoveAll returns
+// syscall.ENOTEMPTY (children has files).
+func reliableRemoveAll(dirPath string) (err error) {
```

Comment:
Can it be simplified similar to this?
```
func reliableRemoveAll(dirPath string) (err error) {
	// Removes all the directories and files.
	err = os.RemoveAll(dirPath)
	if err != nil && isSysErrNotEmpty(err) {
		err = os.RemoveAll(dirPath)
	}
	return err
}
```

Reply:
The reason it is kept the style right now is to allow for future changes i.e i > 1 - it indicates that it is configurable in future.

Reply:
ok